x-pack/filebeat/input/gcs/input_test.go

```diff
@@ -78,24 +86,213 @@ func Test_StorageClient(t *testing.T) {
+			isError:         errors.New("storage: bucket doesn't exist"),
+			unexpectedError: nil,
+		},
+		{
+			name: "Test8_SingleBucketWithPoll_InvalidConfigValue",
+			baseConfig: map[string]interface{}{
+				"project_id":                 "elastic-sa",
+				"auth.credentials_file.path": "/gcs_creds.json",
+				"max_workers":                5100,
+				"poll":                       true,
+				"poll_interval":              "5s",
+				"parse_json":                 false,
+				"buckets": []map[string]interface{}{
+					{
+						"name": "gcs-test",
+					},
+				},
+			},
+			mockHandler:     mock.GCSServer,
+			expected:        map[string]bool{},
+			isError:         errors.New("requires value <= 5000 accessing 'max_workers'"),
+			unexpectedError: nil,
+		},
+		{
+			name: "Test9_TwoBucketWithPoll_InvalidConfigValue",
+			baseConfig: map[string]interface{}{
+				"project_id":                 "elastic-sa",
+				"auth.credentials_file.path": "/gcs_creds.json",
+				"max_workers":                5100,
+				"poll":                       true,
+				"poll_interval":              "5s",
+				"parse_json":                 false,
+				"buckets": []map[string]interface{}{
+					{
+						"name": "gcs-test",
+					},
+					{
+						"name": "gcs-latest",
+					},
+				},
+			},
+			mockHandler:     mock.GCSServer,
+			expected:        map[string]bool{},
+			isError:         errors.New("requires value <= 5000 accessing 'max_workers'"),
+			unexpectedError: nil,
+		},
+		{
+			name: "Test10_SingleBucketWithPoll_parseJSON",
+			baseConfig: map[string]interface{}{
+				"project_id":                 "elastic-sa",
+				"auth.credentials_file.path": "/gcs_creds.json",
+				"max_workers":                1,
+				"poll":                       true,
+				"poll_interval":              "5s",
+				"parse_json":                 true,
+				"buckets": []map[string]interface{}{
+					{
+						"name": bucketGcsTestLatest,
+					},
+				},
+			},
+			mockHandler: mock.GCSServer,
+			checkJSON:   true,
+			expected: map[string]bool{
+				mock.Gcs_test_latest_object_ata_json_parsed:   true,
+				mock.Gcs_test_latest_object_data3_json_parsed: true,
+			},
+			unexpectedError: context.Canceled,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			serv := httptest.NewTLSServer(tt.mockHandler())
+			serv := httptest.NewServer(tt.mockHandler())
 			httpclient := http.Client{
 				Transport: &http.Transport{
 					TLSClientConfig: &tls.Config{
 						InsecureSkipVerify: true, //nolint:gosec // We can ignore as this is just for testing
 					},
 				},
 			}
+			t.Cleanup(serv.Close)
+
 			client, _ := storage.NewClient(context.Background(), option.WithEndpoint(serv.URL), option.WithoutAuthentication(), option.WithHTTPClient(&httpclient))
 			cfg := conf.MustNewConfigFrom(tt.baseConfig)
 			conf := config{}
-			assert.NoError(t, cfg.Unpack(&conf))
+			err := cfg.Unpack(&conf)
+			if err != nil {
+				assert.EqualError(t, err, tt.isError.Error())
+			}
```

Comment:
This doesn't check that `Unpack` correctly returns an error when expected, only that if it does it is the expected error.

If `err != nil` you should return because there is no more work to do.

Reply:
There are a test cases that check for a config error when invalid values, values above max threshold are given. Hence this block is there for those use cases.